okhttp/src/main/java/okhttp3/internal/connection/StreamAllocation.java

```diff
@@ -201,7 +201,7 @@ private RealConnection findConnection(int connectTimeout, int readTimeout, int w
 
   public void streamFinished(boolean noNewStreams, HttpCodec codec) {
     synchronized (connectionPool) {
-      if (codec == null || codec != this.codec) {
+      if (codec != this.codec) {
```

Comment:
This seems completely unrelated to the change. Please submit it as a separate PR.